From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kirill Yatsenko <kiriyatsenko@gmail.com>
Date: Thu, 03 Apr 2026 15:00:00 +0200
Subject: dts: Add device tree for decktrix custom PCB

Decktrix v1 custom PCB based on STM32MP157F.
LDO1 at 2.8V for display AVDD, LDO6 at 1.8V for display IOVCC.

Signed-off-by: Kirill Yatsenko <kiriyatsenko@gmail.com>
---
 fdts/decktrix-v1.dts           | 47 ++++++++++++++++++++++++++++++++
 fdts/decktrix-v1-fw-config.dts |  7 ++++
 2 files changed, 54 insertions(+)
 create mode 100644 fdts/decktrix-v1.dts
 create mode 100644 fdts/decktrix-v1-fw-config.dts

diff --git a/fdts/decktrix-v1.dts b/fdts/decktrix-v1.dts
new file mode 100644
index 00000000..3b4dfd1f
--- /dev/null
+++ b/fdts/decktrix-v1.dts
@@ -0,0 +1,47 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR BSD-3-Clause)
+/*
+ * Copyright (C) Decktrix Lab
+ * Author: Kirill Yatsenko <kiriyatsenko@gmail.com>
+ *
+ * Decktrix v1 custom PCB based on STM32MP157F.
+ * I2C4 pinctrl configured via patch in stm32mp15-pinctrl.dtsi.
+ */
+
+/dts-v1/;
+
+#include "stm32mp157.dtsi"
+#include "stm32mp15xc.dtsi"
+#include "stm32mp15-pinctrl.dtsi"
+#include "stm32mp15xxac-pinctrl.dtsi"
+#include "stm32mp15xx-dkx.dtsi"
+
+/ {
+	model = "Decktrix v1";
+	compatible = "decktrix,v1", "st,stm32mp157";
+
+	aliases {
+		serial3 = &usart2;
+	};
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+	};
+};
+
+&cryp1 {
+	status = "okay";
+};
+
+/* PMIC LDO1: 2.8V for display AVDD (2V8 rail on decktrix PCB) */
+&v1v8_audio {
+	regulator-name = "ldo1";
+	regulator-min-microvolt = <2800000>;
+	regulator-max-microvolt = <2800000>;
+};
+
+/* PMIC LDO6: 1.8V for display IOVCC (1V8 rail on decktrix PCB) */
+&v1v2_hdmi {
+	regulator-name = "ldo6";
+	regulator-min-microvolt = <1800000>;
+	regulator-max-microvolt = <1800000>;
+};
diff --git a/fdts/decktrix-v1-fw-config.dts b/fdts/decktrix-v1-fw-config.dts
new file mode 100644
index 00000000..346fc98f
--- /dev/null
+++ b/fdts/decktrix-v1-fw-config.dts
@@ -0,0 +1,7 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR BSD-3-Clause)
+/*
+ * Copyright (C) Decktrix Lab
+ */
+
+#define DDR_SIZE	0x20000000 /* 512MB */
+#include "stm32mp15-fw-config.dtsi"
--
2.43.0

